### PR TITLE
feat: horoscope command

### PR DESCRIPTION
**Pull request by favna**
[image](https://discordapp.com/channels/254360814063058944/257561807500214273/696218779167883325)

### CODE REVIEW

File: src/commands/Tools/Websearch/horoscope.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	usageDelim: ' '
+})
+export default class extends SkyraCommand {
+
+	private readonly kSunSigns = new Set(['capricorn', 'aquarius', 'pisces', 'aries', 'taurus', 'gemini', 'cancer', 'leo', 'virgo', 'libra', 'scorpio', 'sagittarius']);
+	private readonly kStringArg = this.client.arguments.get('string')!;
+
+	public async init() {
+		this.createCustomResolver('sunsign', (arg, possible, message) => {
+			if (this.kSunSigns.has(arg.toLowerCase())) return this.kStringArg.run(arg, possible, message);
```
Comment (kyranet) [inline src/commands/Tools/Websearch/horoscope.ts]: Why does this need to run the string argument? It's already been validated...

File: src/languages/en-US.ts [hunk 2]
```diff
@@ -3423,6 +3423,29 @@ export default class extends Language {
...
+		COMMAND_HOROSCOPE_DESCRIPTION: 'Get your latest horoscope',
+		COMMAND_HOROSCOPE_EXTENDED: builder.display('horoscope', {
+			extendedHelp: 'Gets the horoscope for a given sun sign from Kelli Fox\'s The Astrologer.',
+			explainedUsage: [
+				['sunsign', 'The sun sign you want to get the horoscope for'],
+				['today|tomorrow|yesterday', '(Optional, defaults to "today") If you want to get the horoscope of yesterday or tomorrow you can specify that.']
+			],
+			examples: ['pisces', 'virgo tomorrow', 'gemini yesterday', 'aries today']
+		}),
+		COMMAND_HOROSCOPE_INVALID_SUNSIGN: sign => `${sign} is an invalid sun sign, maybe try ${createPick([
```
Comment (kyranet) [inline src/languages/en-US.ts]: Maybe make an utility that accesses to a random entry and make createPick use it? So we avoid allocating a callback every time this is run (plus `createPick([])()` looks weird)

File: src/languages/es-ES.ts [hunk 3]
```diff
@@ -3436,6 +3436,29 @@ export default class extends Language {
...
+			'gemini', 'cancer', 'leo', 'virgo', 'libra', 'scorpio',
+			'sagittarius'
+		])()}`,
+		COMMAND_HOROSCOPE_TITLES: {
+			DAILY_HOROSCOPE: sign => `Horóscopo diario para ${sign}`,
+			METADATA_TITLE: 'Metadatos',
+			METADATA: (intensity, keywords, mood) => ([
+				`**Intensidad:** ${intensity}`,
+				`**Palabras clave:** ${keywords}`,
+				`**Estado animico:** ${mood}`
```
Comment (kyranet) [inline src/languages/es-ES.ts]:
```suggestion
				`**Estado anímico:** ${mood}`
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/commands/Tools/Websearch/horoscope.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	description: language => language.tget('COMMAND_HOROSCOPE_DESCRIPTION'),
+	extendedHelp: language => language.tget('COMMAND_HOROSCOPE_EXTENDED'),
+	requiredGuildPermissions: ['EMBED_LINKS'],
+	usage: '<sunsign:sunsign> [tomorrow|yesterday|today:default]',
+	usageDelim: ' '
+})
+export default class extends SkyraCommand {
+
+	private readonly kSunSigns = new Set(['capricorn', 'aquarius', 'pisces', 'aries', 'taurus', 'gemini', 'cancer', 'leo', 'virgo', 'libra', 'scorpio', 'sagittarius']);
+	private kRandomSunSign = createPick([...this.kSunSigns]);
```
Comment (kyranet) [inline src/commands/Tools/Websearch/horoscope.ts]:
```suggestion
	private readonly kRandomSunSign = createPick([...this.kSunSigns]);
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/commands/Tools/Websearch/horoscope.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+	public async init() {
+		this.createCustomResolver('sunsign', (arg, _, message) => {
+			if (this.kSunSigns.has(arg.toLowerCase())) return arg;
```
Comment (kyranet) [inline src/commands/Tools/Websearch/horoscope.ts]: Does the horoscope support signs in case insensitive? Just checking whether or not this is valid

Comment (favna) [reply src/commands/Tools/Websearch/horoscope.ts]: oh shit good catch, because it doesn't.

File: src/commands/Tools/Websearch/horoscope.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	usageDelim: ' '
+})
+export default class extends SkyraCommand {
+
+	private readonly kSunSigns = new Set(['capricorn', 'aquarius', 'pisces', 'aries', 'taurus', 'gemini', 'cancer', 'leo', 'virgo', 'libra', 'scorpio', 'sagittarius']);
+	private readonly kRandomSunSign = createPick([...this.kSunSigns]);
+
+	public async init() {
+		this.createCustomResolver('sunsign', (arg, _, message) => {
+			if (this.kSunSigns.has(arg.toLowerCase())) return arg.toLowerCase();
```
Comment (kyranet) [inline src/commands/Tools/Websearch/horoscope.ts]:
```suggestion
			const lowerCasedArgument = arg.toLowerCase();
			if (this.kSunSigns.has(lowerCasedArgument)) return lowerCasedArgument;
```

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Pull request merged**